google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -227,8 +230,11 @@ public CloudSdk createSdk(
         .appCommandCredentialFile(credentialsPath.toFile())
         .appCommandMetricsEnvironment(pluginInfoService.getExternalPluginName())
         .appCommandMetricsEnvironmentVersion(pluginInfoService.getPluginVersion())
-        .appCommandOutputFormat("json")
-        .build();
+        .appCommandOutputFormat("json");
+
+    getProjectJavaSdk(project).ifPresent(sdkBuilder::javaSdkPath);
```

Comment:
From the point of view of dissociating the API from its internals, which do you think makes more sense -- specifying the JDK at Cloud SDK creation time or at command config time?

Thinking about a day we would bypass the Cloud SDK and have a pure Java API, having this dependency at the command level would stop making sense. From this point of view, the JDK location would be a dependency of the Cloud SDK and not of the command.
Also, the Cloud SDK path is another such example -- it's not used for devappserver commands and yet we still define it at the Cloud SDK level.